python/lsst/pipe/tasks/characterizeImage.py

```diff
@@ -0,0 +1,493 @@
+            mask is updated accordingly, and the PSF model is set
+        - sourceCat  detected sources (an lsst.afw.table.SourceCatalog)
+        - background  model of background subtracted from exposure (an lsst.afw.math.BackgroundList)
+        - psfCellSet  spatial cells of PSF candidates (an lsst.afw.math.SpatialCellSet)
+        """
+        # install a simple PSF model, if wanted
+        if self.config.doMeasurePsf:
+            if self.config.useSimplePsf or not exposure.hasPsf():
+                self.installSimplePsf.run(exposure=exposure)
+        elif not exposure.hasPsf():
+            raise pipeBase.TaskError("exposure has no PSF model and config.doMeasurePsf false")
+
+        # run repair, but do not interpolate over cosmic rays (do that elsewhere, with the final PSF model)
+        self.repair.run(exposure=exposure, keepCRs=True)
+        self.display("repair_iter", exposure=exposure)
+
+        damRes = self.detectAndMeasure.run(
+            exposure = exposure,
+            exposureIdInfo = exposureIdInfo,
+            background = background,
+            allowApCorr = False,  # wait for final PSF
+        )
+
+        measPsfRes = pipeBase.Struct(
+            cellSet = None
+        )
+        if self.config.doMeasurePsf:
+            if self.measurePsf.usesMatches:
+                matches = self.astrometry.loadAndMatch(
+                    exposure = exposure,
+                    sourceCat = damRes.sourceCat,
+                ).matches
+            else:
+                matches = None
+            measPsfRes = self.measurePsf.run(
+                exposure = exposure,
+                sources = damRes.sourceCat,
+                matches = matches,
+            )
+        self.display("measure_iter", exposure=exposure, sourceCat=damRes.sourceCat)
+
+        return pipeBase.Struct(
+            exposure = exposure,
+            sourceCat = damRes.sourceCat,
+            background = damRes.background,
+            psfCellSet = measPsfRes.cellSet,
+        )
+
+    def getSchemaCatalogs(self):
+        """Return a dict of empty catalogs for each catalog dataset produced by this task.
+        """
+        sourceCat = SourceCatalog(self.schema)
+        sourceCat.getTable().setMetadata(self.detectAndMeasure.algMetadata)
+        return {"icSrc": sourceCat}
+
+    def display(self, itemName, exposure, sourceCat=None):
+        """Display exposure and sources on next frame, if display of itemName has been requested
+
+        @param[in] itemName  name of item in debugInfo
+        @param[in] exposure  exposure to display
+        @param[in] sourceCat  source catalog to display
+        """
+        val = getDebugFrame(self._display, itemName)
+        if not val:
+            return
+
+        displayAstrometry(
+                exposure = exposure,
+                sourceCat = sourceCat,
+                frame = self._frame,
+                pause = False,
```

Comment:
I think you do want the `pause`, because of the iteration and the many things being displayed.  Otherwise, the user's going to lose track of what's being displayed.  I think there also needs to be some title on the display to help disambiguate.